### PR TITLE
Restructure documentation

### PR DESCRIPTION
**Pull request by jrudolph**
On my yak-shaving trip, I ended up not finding the information I was looking for in the Paradox documentation so I restructured it.

The TOC was reorganized by these rules:

 * Information to *use* paradox on the top-level
 * All directives information in one place
 * Customization info in an extra subitem

Mostly, existing content was just moved around:
 * There's a new configuration section, which got content of smaller sub-pages (still needs documentation of available configuration settings)
 * Added sections for "Variable Substitution" and the `@var`/`@@@vars` directives
 * Added a simple index of directives (could be made nicer later)
 * Created a new "Getting Started" page that got content from the main index
 * Moved "Theming" and "Templating" pages to a new section "Extensions", which now also has an index of known extensions / themes for Paradox (I find the scope of Theming and Templating still confusing, could need an overhaul)

### New Index Page and TOC
new-paradox-toc

### New Directives Index

directives-structure

### CODE REVIEW

File: docs/src/main/paradox/features/version-warning.md [hunk 1]
```diff
@@ -1,12 +0,0 @@
-Version warning
```
Comment (jrudolph) [inline docs/src/main/paradox/features/version-warning.md]: Moved to configuration

File: docs/src/main/paradox/features/overlay.md [hunk 2]
```diff
@@ -1,23 +0,0 @@
-Documentation overlay
```
Comment (jrudolph) [inline docs/src/main/paradox/features/overlay.md]: Moved to configuration

File: docs/src/main/paradox/features/multi-configuration.md [hunk 3]
```diff
@@ -1,21 +0,0 @@
-Multi Configuration
```
Comment (jrudolph) [inline docs/src/main/paradox/features/multi-configuration.md]: Moved to configuration

File: docs/src/main/paradox/index.md [hunk 4]
```diff
@@ -1,73 +1,20 @@
-Paradox
-=======
+# Paradox
 
 Paradox is a Markdown documentation tool for software projects.
-The GitHub repo is [lightbend/paradox][repo].
 
-**Paradox is NOT supported under the Lightbend subscription.**
-
-### Setup
```
Comment (jrudolph) [inline docs/src/main/paradox/index.md]: Moved to "Getting Started"

File: docs/src/main/paradox/index.md [hunk 5]
```diff
@@ -1,73 +1,20 @@
...
-    paradoxTheme := Some(builtinParadoxTheme("generic"))
-  )
-```
-
-Then call `paradox` which will generate the site in `target/paradox/site/main`.
-
-Your markdown documentation will go inside `src/main/paradox/`. For example, you can start with `src/main/paradox/index.md`.
+@@toc { depth=1 }
 
-### Key features
```
Comment (jrudolph) [inline docs/src/main/paradox/index.md]: Moved to new "Overview" page

File: project/plugins.sbt [hunk 6]
```diff
@@ -20,7 +20,7 @@ addSbtPlugin("de.heikoseeberger"     % "sbt-header"      % "5.0.0")
 addSbtPlugin("org.foundweekends"     % "sbt-bintray"     % "0.5.1")
 addSbtPlugin("org.xerial.sbt"        % "sbt-sonatype"    % "2.0")
 addSbtPlugin("com.jsuereth"          % "sbt-pgp"         % "1.1.0")
-addSbtPlugin("com.lightbend.paradox" % "sbt-paradox"     % "0.3.4")
+addSbtPlugin("com.lightbend.paradox" % "sbt-paradox"     % "0.3.6")
```
Comment (jrudolph) [inline project/plugins.sbt]: Updated to be able to use absolute snippet paths.

**Comment by ktoso** [comment]
Looks good to me :)

File: build.sbt [hunk 7]
```diff
@@ -134,6 +134,8 @@ lazy val docs = (project in file("docs"))
       "empty" -> "",
       "version" -> version.value
     ),
+    paradoxNavigationDepth := 1,
```
Comment (ktoso) [inline build.sbt]: would you consider 2 for quicker navigation?

Comment (jrudolph) [reply build.sbt]: I guess it could make sense:


toc-depth2

Comment (ktoso) [reply build.sbt]: Yes plz

Comment (jrudolph) [reply build.sbt]: Done

**Comment by ktoso** [comment]
Anyone wanna veto merging this? I think it looks great :-)

**Comment by jrudolph** [comment]
Should get 0.4.0 milestone (but I'm not an owner here).

**Pull request merged**